docs/userguide/quickstart.rst

```diff
@@ -186,17 +186,17 @@ For more details, see :doc:`datafiles`
 
 Development mode
 ================
-``setuptools`` allows you to install a package without copying any files
-to your interpreter directory (e.g. the ``site-packages`` directory). This
-allows you to modify your source code and have the changes take effect without
-you having to rebuild and reinstall. This is currently incompatible with
-PEP 517 and therefore it requires a ``setup.py`` script with the following
-content::
 
-    import setuptools
-    setuptools.setup()
+.. tip::
+
+	When there is no ``setup.py`` script present, this is only
+	compatible with :pep:`517` since :ref:`pip v21.1 <pip:v21-1>`.
```

Comment:
nit: this _what_? This section? This mode of development? This tip? I've found that use of "this" without an explicit reference is often ambiguous. It's common to see such usage, but I avoid it.

I'd probably rewrite this tip as:

```suggestion
	Prior to :ref:`pip v21.1 <pip:v21-1>`, a ``setup.py`` script was
	required to be compatible with development mode. With late
	versions of pip, any project may be installed in this mode.
```

Reply:
You're right. Your wording is much better. Thanks! This is why I like code reviews — so that somebody could refactor my brain dump a bit :)